protoc-gen-grpc-gateway/descriptor/grpc_api_configuration.go

```diff
@@ -0,0 +1,94 @@
+package descriptor
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+// GrpcAPIService represents a stripped down version of google.api.Service .
+// Compare to https://github.com/googleapis/googleapis/blob/master/google/api/service.proto
+//
+// Note that for the purposes of the gateway generator we only consider a subset of all
+// available features google supports in their service descriptions and hence we do not
+// bother having the full service protobuf which has a lot of other dependencies. Thanks
+// to backwards compatibility this should be relatively safe.

```

Comment:
"Thanks to backwards compatibility" doesn't seem to follow from the rest of the comment. It seems like "thanks to structural typing" may be a better way to phrase that?